### DISCUSSION TITLE
How would i implement my own componentlibrary

### DISCUSSION THREAD

**Discussion by luckyluggi** [microsoft/fluentui-blazor#2312] [Q&A] [answered]
My company uses blazor and we want to create a component library maintained by one of our teams and used by all other teams.
- The designsystem will probably need to feature most components that fluentui-blazor offers, but not all.
- It will also feature additional components that might be composed of multiple fluentui-blazor components.
- And it should also restrict our teams to not use all features of the fluentui-blazor components (for example we'd like to unly use the filled appearance for the button)

When i look at fast.design's documentation they have clear guidelines for every usecase here:
https://www.fast.design/docs/introduction/#where-should-i-start

How would i build my library with fluientui-blazor?
Would i create a MyButton thet has its own properties and html and redirect the props to the fluentbutton like this?

<FluentButton Appearance=@Appearance .Filled Required=@Required ... />

Or would you say it is better to not wrap the components and use them directly (not having our own restricted library)

**Comment by vnbaaij** [comment, marked as answer]
Hi,

With Blazor you'd normally use the components as-is. It is not impossible to wrap them but I don't see any compelling reasons to do so. It will be very hard to style things, etc. 

You choose to use this library because you want your app to have the Fluent Design look and feel. If you want to deviate from that design, that is fine of course, but then this library is probably not the best choice for you.

**Reply by luckyluggi** [reply]
Ok i understand. 
I have a followup question. 
We use Micro Frontends and might also enable other companies to create micro frontends using react. We'd like to have the same components available there to keep a consistent look and feel. 

To make this possible i think it might be best to have the componentlibrary build as webcomponents. I guess i'd create such a library using just fast.design. Am i right?
But looking at fast.design itself there are way less components than in the fluentui-blazor. Is fast.design still maintained?

**Reply by vnbaaij** [reply]
You're a bit in the wrong corner here as this repo is only about the Fluent UI Blazor components. We are totally separate from the Fluent UI React components. We share the same design but have different components/implementations.

FAST is the technical underpinning on which the web components are built. There was a FAST components offering earlier, but that has been deprecated. FAST itself is still being built and maintained. 

Go with the Fluent UI Web Components to be future proof. The current version is v2 and the team is hard at work at building v3 (which will support Fluent Design v2)

**Reply by luckyluggi** [reply]
Thanks
